cmd/ingester/app/consumer/consumer.go

```diff
@@ -99,7 +103,18 @@ func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 	msgMetrics := c.newMsgMetrics(pc.Partition())
 	var msgProcessor processor.SpanProcessor
 
-	for msg := range pc.Messages() {
+	// "Cast" channel from chan *sarama.ConsumerMessage to chan interface{}.
```

Comment:
Could you elaborate which problem we are solving here? 
The casting seems to happen on L117, and L107 is creating a new channel to copy all messages.

Reply:
Definitely will add more detail to comments. A channel such as `chan *sarama.ConsumerMessage` cannot be cast to `chan interface{}`, so I need to forward from the source to the destination channel.